Задачи после прослушивания 4 вебинара

RumyantsevP opened the pull request:
Добавил решенные задачи после прослушивания вебинара №4

Commit:
Задачи после прослушивания 4 вебинара

DevanovAuto commented on lesson_4/task1.py:
```diff
@@ -0,0 +1,39 @@
+    if (a + b > c) and (a + c > b) and (b + c > a):  # Проверяем условие существования треугольника по трем сторонам
+        if a == b == c:  # Если все стороны треугольника равны, то треугольник "Равносторонний"
+            type_triangle = 'Равносторонний'
+        elif (a == b) or (b == c) or (a == c):  # Если две стороны треугольника равны, то треугольник "Равнобедренный"
```
скобки необязательны тут

RumyantsevP replied:
Ок, скорректирую, уберу скобки в 12 строке

DevanovAuto commented on lesson_4/task2.py:
```diff
@@ -0,0 +1,35 @@
+def flatten_and_sort(array):
+    # Здесь нужно написать код
+    result_list = []  # Заводим пустой список, в который в дальнейшем будем класть значения
+    for j in data[i]:  # Пробегаем по исходному списку и получаем вложенный список
```
список и так приходит в функцию, зачем обращаться к data?

RumyantsevP replied:
Скорректировал у себя на компьютере код, чтобы не обращаться к "data", а сразу обращаемся к "array". Тесты проходят

DevanovAuto commented on lesson_4/task1.py:
```diff
@@ -0,0 +1,39 @@
+            type_triangle = 'Равносторонний'
+        elif (a == b) or (b == c) or (a == c):  # Если две стороны треугольника равны, то треугольник "Равнобедренный"
+            type_triangle = 'Равнобедренный'
+        elif a != b != c:  # Если все стороны треугольника не равны, то треугольник "Обычный"
```
тут можно else без условия

RumyantsevP replied:
Не понимаю, что здесь не так. Здесь проверяем в elif является ли треугольник "Обычным". Блок else ниже, без условия

DevanovAuto commented on lesson_4/task4.py:
```diff
@@ -0,0 +1,37 @@
+        count_multy += 1  # Увеличиваем количество раз, сколько раз перемножаются цифры числа
+        for i in num:  # Пробегаем по цифрам числа
+            new_number *= int(i)  # Перемножаем цифры числа между собой
+            num = str(new_number)  # Переводим полученное число в формат строки
```
логичнее вынести из цикла

DevanovAuto commented on lesson_4/task6.py:
```diff
@@ -0,0 +1,33 @@
+def create_phone_number(num_tuple):
+    # Здесь нужно написать код
+    old_str_phone = ''  # Заводим пустую строку, в которую будут укладываться предварительные данные
+    for j in data[i]:  # Перебираем элементы кортежа
```
надо итерироваться по num_tuple

DevanovAuto commented on lesson_4/task6.py:
```diff
@@ -0,0 +1,33 @@
+    for j in data[i]:  # Перебираем элементы кортежа
+        old_str_phone += str(j)  # Складываем элементы кортежа в промежуточную строку
+    # Формируем итоговую строку, в которую добавляем элементы из промежуточной строки с добавлением разделителей для формирования номера телефона
+    str_phone = "(" + old_str_phone[0] + old_str_phone[1:3] + ")" + " " + old_str_phone[3:6] + "-" + old_str_phone[6:]
```
почему сразу не написать old_str_phone[:3]?

DevanovAuto commented on lesson_4/task7.py:
```diff
@@ -0,0 +1,40 @@
+    # Здесь нужно написать код
+    null_list = []  # Заводим список, в который будем складывать только нули из исходного списка (lst)
+    lst = []  # Заводим список, в который будем складывать не нулевые значения из исходного списка (lst)
+    for j in data[i]:  # Перебираем значения в списке
```
итерируемся по параметру функции

RumyantsevP committed:
Update task1.py

Поправил код в соответствии с замечаниями

RumyantsevP committed:
Update task2.py

Скорректировал код, чтобы обращаться не к "data", а сразу к "array"

RumyantsevP committed:
Update task1.py

Поправил блок elif, в котором проверяется является ли треугольник "Обычным"

RumyantsevP committed:
Update task4.py

Вынес из цикла перевод полученного числа в формат строки (код строка 19)

RumyantsevP committed:
Update task6.py

1. Итерируемся по num_tuple, а не по data (код - строка 9)
2. Скорректировал сбор итоговой строки с номером телефона (код - строка 12)

RumyantsevP committed:
Update task7.py

Итерируемся не по data, а по параметру функции (lst)
